### PR TITLE
Add support for serde

### PR DESCRIPTION
**Pull request by hpeebles**

### CODE REVIEW

**Comment by spearman** [comment]
I'm curious why you didn't use serde derive?

**Comment by hpeebles** [comment]
> I'm curious why you didn't use serde derive?

In order to use derive we would need to add a trait bound on `A`, see the example below -

```
#[derive(Clone,Debug,Eq,PartialEq)]
#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
pub struct RangeSet <A> where
  A       : smallvec::Array + Eq + std::fmt::Debug,
  A::Item : Clone + Eq + std::fmt::Debug
{
  ranges  : smallvec::SmallVec <A>
}
```

```
error[E0277]: the trait bound `<A as Array>::Item: Serialize` is not satisfied
    --> src/lib.rs:56:3
     |
56   |   ranges  : smallvec::SmallVec <A>
     |   ^^^^^^ the trait `Serialize` is not implemented for `<A as Array>::Item`
     |
     = note: required because of the requirements on the impl of `Serialize` for `SmallVec<A>`
```

However I have just pushed an update which enables the `serde` feature on `smallvec` which means we can simply call through into that implementation.

File: Cargo.toml [hunk 1]
```diff
@@ -9,4 +9,8 @@ readme = "README.md"
 
 [dependencies]
 num-traits = "0.2.*"
-smallvec = "0.6.*"
+serde = { version = "1.0.*", optional = true }
+smallvec = { version = "0.6.*" }
+
+[features]
+serde = ["dep:serde", "smallvec/serde"]
```
Comment (hpeebles) [inline Cargo.toml]: Ahh hold on this syntax requires Rust 1.60.0, will see what can be done to support older versions.

Comment (hpeebles) [reply Cargo.toml]: https://github.com/rust-lang/api-guidelines/discussions/180
I'll have a play around with some of the suggestions here later.

Comment (hpeebles) [reply Cargo.toml]: I have now changed it to be compatible with old versions of Rust.

**Comment by spearman** [comment]
Thanks, published in v0.0.9

**Pull request merged**